### DISCUSSION TITLE
about the 'mlx_put_string'

### DISCUSSION THREAD

**Discussion by ThEWoLf882-42** [codam-coding-college/MLX42#97] [Q&A] [answered]
i am using mlx_put_string and i am wondering how can i change string color and size
(font maybe)

**Comment by W2Wizard** [comment, marked as answer]
The color could be potentially be changed by for instance applying some sort of multiplication to each pixel. The size however is fixed, font's are notorousliy difficult to handle.

If you truly want to be able to have your own font besides the default one given by MLX, you have to implement it on your own. Using a sprite sheet is a good start and then creating an image with the correct width and height to which you can copy paste the pixels to should work.

**Reply by Obult** [reply]
Multiplication for color change per pixel? Or did you mean color is fixed and the size can be changed that way?

**Reply by JellevanKraaij** [reply]
It looks like the font that the current put_string uses internally is basically just a texture where the function just calculates where the character is located and then just copies that section to the window. The color could be changed by making a alternative copy function that changes color of the nonzero pixels. But in that case it would be better to use a fond definition that only contains which pixel should be colored and which pixel should be black.

The only good solution to size would be to include different sizes of the same font, scaling fonts generally are ugly and doesn't improve the readability.

**Reply by JellevanKraaij** [reply]
The biggest problem would probably be the antialiasing with different colours, maybe just don't do antialiasing and use a font type that looks good without.
